crates/pathfinder/src/state/merkle_tree.rs

```diff
@@ -1570,6 +1651,566 @@ mod tests {
+
+            let value_1 = starkhash!("02");
+            let value_2 = starkhash!("03");
+            let value_3 = starkhash!("05");
+
+            uut.set(key1, value_1).unwrap();
+            uut.set(key2, value_2).unwrap();
+            uut.set(key3, value_3).unwrap();
+
+            let root = uut.commit().unwrap();
+            let uut = MerkleTree::load("test", &transaction, root).unwrap();
+
+            let proofs = get_proofs(&keys, &uut).unwrap();
+            let verified_1 = verify_proof(root, key1, value_1, &proofs[0]).unwrap();
+            assert_eq!(verified_1, Membership::Member, "Failed to prove key1");
+
+            let verified_2 = verify_proof(root, key2, value_2, &proofs[1]).unwrap();
+            assert_eq!(verified_2, Membership::Member, "Failed to prove key2");
+
+            let verified_key3 = verify_proof(root, key3, value_3, &proofs[2]).unwrap();
+            assert_eq!(verified_key3, Membership::Member, "Failed to prove key3");
+        }
+
+        #[test]
+        fn left_edge() {
+            let mut conn = rusqlite::Connection::open_in_memory().unwrap();
+            let transaction = conn.transaction().unwrap();
+            let mut uut = MerkleTree::load("test", &transaction, StarkHash::ZERO).unwrap();
+
+            //  (251,0x00,0x99)
+            //       /
+            //      /
+            //   (0x99)
+
+            let key_1 = starkhash!("00"); // 0b00
+
+            let key1 = key_1.view_bits();
+            let keys = [key1];
+
+            let value_1 = starkhash!("aa");
+
+            uut.set(key1, value_1).unwrap();
+
+            let root = uut.commit().unwrap();
+            let uut = MerkleTree::load("test", &transaction, root).unwrap();
+
+            let proofs = get_proofs(&keys, &uut).unwrap();
+            let verified_1 = verify_proof(root, key1, value_1, &proofs[0]).unwrap();
+            assert_eq!(verified_1, Membership::Member, "Failed to prove key1");
+        }
+
+        #[test]
+        fn left_right_edge() {
+            let mut conn = rusqlite::Connection::open_in_memory().unwrap();
+            let transaction = conn.transaction().unwrap();
+            let mut uut = MerkleTree::load("test", &transaction, StarkHash::ZERO).unwrap();
+
+            //  (251,0xff,0xaa)
+            //     /
+            //     \
+            //   (0xaa)
+
+            let key_1 = starkhash!("ff"); // 0b11111111
+
+            let key1 = key_1.view_bits();
+            let keys = [key1];
+
+            let value_1 = starkhash!("aa");
+
+            uut.set(key1, value_1).unwrap();
+
+            let root = uut.commit().unwrap();
+            let uut = MerkleTree::load("test", &transaction, root).unwrap();
+
+            let proofs = get_proofs(&keys, &uut).unwrap();
+            let verified_1 = verify_proof(root, key1, value_1, &proofs[0]).unwrap();
+            assert_eq!(verified_1, Membership::Member, "Failed to prove key1");
+        }
+
+        #[test]
+        fn right_most_edge() {
```

Comment:
Technically this and the `left_right_edge` test is the same test case as `left_edge`, except for the fact that the edge node, which is the root node contains a path with some or all bits set.

I'm not sure if we should keep all of the 3 tests, however the ascii art is pretty illustrative in how one could visualize those cases and I really like it.

Reply:
Haha not gonna lie the ASCII art was here for my personal understanding as it was the first time for me working with MPTs.

If you want to remove some redundant tests we can. I think it doesn't hurt leaving them, but I won't take it personally if we remove them 😂